File: study-scala/src/main/scala/com/ffl/study/scala/test/HelloWorld.java
package com.ffl.study.scala.test;

/**
 * @author lff
 * @datetime 2020/05/20 01:46
 *
 * 理解为scala在运行时做了一个包装
 */
public class HelloWorld {

    public static void main(String[] args) {
        HelloWorld$.MODULE$.main(args);
    }
}


final class HelloWorld$ {

    public static final HelloWorld$ MODULE$;

    static {
        MODULE$ = new HelloWorld$();
    }

    public void main(String[] args){
        System.out.println("hello,scala, idea...");
    }
}

File: study-scala/src/main/scala/com/ffl/study/scala/HelloWorld.scala
package com.ffl.study.scala

/**
  * @author lff
  * @datetime 2020/04/08 13:54
  *
  * 1.object TestScala对应的是一个 TestScala$的一个静态对象 MODULE$
  * 2.在程序中是一个单例
  */
object HelloWorld {

    def main(args: Array[String]): Unit = {
        println("hello scala")


        val num:Int = 10

        println(num.toDouble)
    }



}
